Functions/QualityControl/xASL_qc_GenerateReport.m

```diff
@@ -0,0 +1,418 @@
+        case 'patients' 
+            line = xASL_sub_PrintPatient(x, figure, line, settings);
+    end  
+end
+
+function  xASL_sub_printPage(json, x, settings)
+
+    clear primaryFig ax  
+    primaryFig = figure('visible', 'off','Units', 'centimeters','Position', [0 0 21 29.7]);
+
+    %% Print the Title
+    logoPath = fullfile(x.opts.MyPath, 'Design/ExploreASL_logoHeader.png');
+    [~] = xASL_sub_PrintImage(logoPath, [], primaryFig, settings, [0 0.96 1 0]);
+
+    %% Print the Footer
+    xASL_sub_PrintText('This report was automatically generated by ExploreASL', primaryFig, [0 0.02 1 0], settings);
+
+    %% Parse JSON and create the file
+    [~] = xASL_sub_parseJson(json, x, primaryFig, [0 0.93 1 0], settings);
+
+    PrintFile = ['xASL_Report_', json.pageIdentifier];
+    PrintPath = fullfile(x.dir.xASLDerivatives, x.SUBJECT, PrintFile); 
+
+    print(primaryFig, PrintPath, '-dpdf', '-bestfit');
+
+    clear primaryFig ax
+end
+
+function xASL_sub_printBlock(input, x, pageFig, settings)
+    position = xASL_str2num(input.position);
+    size = xASL_str2num(input.size);
+    [settings.canvas, line] = xASL_sub_createNewCanvas(position, size, settings.canvas);
+    xASL_sub_parseJson(input, x, pageFig, line, settings);
+end
+
+function line = xASL_sub_PrintPatient(x, figure, line, settings)
+    %Check if participant data exists.
+    ExistsParticipants = xASL_adm_GetFileList(x.D.ROOT, 'participants.tsv');
+    PatientInfo = {};
+
+    if isempty(ExistsParticipants)
+        return
+    end
+
+    sParticipants = xASL_tsvRead(ExistsParticipants{1});
+    nParticipants = size(sParticipants, 1);
+    for iPar=2:nParticipants
+        if sParticipants{iPar, 1} == x.SUBJECT
+            %Extract Participants table, and participants info
+            PatientInfo = [sParticipants(1,:); sParticipants(iPar, :)];
+        end
+    end
+
+    if isempty(PatientInfo)
+        fprintf ('no patient info found to be printed \n');
+        return
+    end
+    
+    line = xASL_sub_PrintText('Participant Information', figure, line, settings );
+    for iEntry=1:size(PatientInfo,2)
+        if strcmp(PatientInfo{1, iEntry},'participant_id')
+            line = xASL_sub_PrintText(['Participant: ', PatientInfo{2, iEntry}], figure, line, settings );
+        elseif strcmp(PatientInfo{1, iEntry},'age')
+            line = xASL_sub_PrintText(['Participant age: ', num2str(PatientInfo{2, iEntry})], figure, line, settings );
+        elseif strcmp(PatientInfo{1, iEntry},'sex')
+            line = xASL_sub_PrintText(['Participant sex: ', PatientInfo{2, iEntry}], figure, line, settings );
+        end
+    end
+
+end
+
+function line = xASL_sub_PrintQC(json, x, figure, line, settings)
+    % Stop if module and field don't exists in output
+    if ~isfield(x.Output, (json.module)) && ~isfield(x.Output.(json.module), json.parameter)
+        return     
+    end
+
+    % Use field specific settings if they exists
+    settings.fontName = 'monospace';
```

Comment:
I thought all settings were in the default config file, but some are in this code? Should we be consistent?

Reply:
Yes, but qc-values specifically I prefer to use a different font to the standard default. 
Monospace ensures that the numbers line up better regardless of what comes before it. 
I have explained it better in the comments.